probe_scraper/transform_probes.py

```diff
@@ -267,15 +268,18 @@ def make_item_defn(definition, commit: Commit):
             "last": commit.reflog_index,
         }
     else:
-        # we've seen this definition, update the `last` commit
+        # we've seen this definition, update the `last` commit and source url
         last_dt = datetime.fromisoformat(definition[DATES_KEY]["last"])
         last_timestamp = last_dt.replace(tzinfo=timezone.utc).timestamp()
         last_reflog = definition[REFLOG_KEY]["last"]
         # use negative last_reflog to match commit.sort_key()
-        if (last_timestamp, -last_reflog) < commit.sort_key():
+        if commit.is_head or (last_timestamp, -last_reflog) < commit.sort_key():
```

Comment:
commit.is_head is a stronger indicator that a commit is latest than our sort key